### DISCUSSION TITLE
How do I cast between symbols and strings?

### DISCUSSION THREAD

**Discussion by glyh** [hylang/hy#2397] [Q&A] [answered]

**Comment by Kodiologist** [comment, marked as answer]
Did you try reading the section "Symbols" in the manual?

Edit for future readers: you can convert a string to a symbol with `hy.models.Symbol`, and you can convert a symbol to a string with `str`.

**Reply by glyh** [reply]
Yeah I do. [Here](https://docs.hylang.org/en/stable/syntax.html#symbols) it's mostly about how to create a symbol and how it mangles stuff. I also read the source code [here](https://github.com/hylang/hy/blob/63ae9785cc3b4b5e4fee25b0ed777a12abf77296/hy/models.py#L217), but I didn't find any sections that related to how to convert.

**Reply by Kodiologist** [reply]
Er, what exactly do think of as the distinction between "converting" or "casting" a string to a symbol and "creating" a symbol from a string?

**Reply by glyh** [reply]
My bad, I know how to create a symbol with a string, but not the other wayaround. And I think there's possibly a bug in the compiler:
```hy
(str 'abc) ; => "abc"
(str '1) ; => "hy.models.Integer(1)"
````
I don't know how exactly it is implemented but I think it should at least be consistent. Otherwise this is a footgun for programmers.

**Reply by Kodiologist** [reply]
Your many low-effort posts (on the discussion board and the issue list) are a waste of my time. If you must post, please put substantially more thought and effort into figuring out the situation, and be more selective in what you post.

**Reply by glyh** [reply]
I don't think many of your answers actually solve my questions. Yeah they are small details but it turns out sometimes there's no way to do it(e.g. [import from future](https://github.com/hylang/hy/discussions/2394) ).

If you think these are noise, I will not post any of similar posts again here. Good luck.

**Reply by Kodiologist** [reply]
You'll always get better answers if you ask better questions. For example, if you'd asked "How do I make a symbol into a string?" here, which is what you'd actually wanted to know, you'd probably have gotten to that sooner (but you did, of course, figure it out on your own in a few minutes, hence this still being the kind of question I'm complaining about). In the other case, you asked "How do I prevent macros from polluting the global namespace via import?", when what it seems you meant to ask is "How do I use an `import __future__` statement in a macro expansion?" (but honestly, I'm still not sure). Those are different questions with different answers. Asking questions is a good habit provided that you're thoughtful about the questions.
